accepted/2020/reproducible-builds.md

```diff
@@ -0,0 +1,238 @@
+
+Also, should package consumers find a zero-day exploit, they usually have no way to patch the binaries themselves. They must find the corresponding repo, get the necessary dependencies for building it, somehow find the correct version of the code, patch it, and then rebuild. Not an easy task under ideal circumstances, much less viable when time is of the essence.
+
+To address these problems, we will create a set of tooling that for each managed binary file in a NuGet package, will be able to enumerate all of the input sources used to create it, validate they are accessible and validate that the resulting binary matches was produced using those sources files. In doing so, we don't need to trust the original build environment, as we're able to independently create the managed file from the input sources.
+
+## Scenarios and User Experience
+
+### Customer Cohorts
+
+- Package consumers who wish to manually check results on their own.
+- Package authors who wish to verify their package is reproducible.
+- Package hosts, such as NuGet.org and Terrapin, who wish to validate packages on upload and present the status as part of overall package/project health.
+
+### End-to-end scenarios
+
+#### Developer or InfoSec that wants to validate on their own
+
+An internal package security team is wants to reduce their supply chain risk to their apps and services. They want to make sure that the binaries being used can be traced to sources so they can run security scans on the source code. The packages being validated are produced from a mix of public and internal sources. The team sets up infrastructure to run the validation tool for each package they use.
+
+Due to organization security requirements, based on likely forthcoming NIST standards, they must be able to trace the source and rebuild all packages being used.
+
+1. For each managed binary in the NuGet package, call a tool to check the status. Result is one of: not buildable, buildable but not verifiable (builds but not the same deterministic output), or deterministically reproducible.
+2. If all of the managed binaries are deterministically reproducible, the package is marked as reproducible/validated.
+
+#### Developer creating a library
+
+A developer is updating their library and wants ensure that it can be successfully validated so their users feel secure in taking a dependency on it, and that NuGet.org will give it a higher health score.
+
+The developer knows that by updating to the latest .NET SDK, they are secure by default because the right settings are automatically enabled for their project:
+
+- Source Link
+- EmbedUntrackedSources
+- PublishRepositoryUrl
+- Latest Roslyn compiler which stores compiler flags into the PDB
+- PDB is automatically included in the NuGet package
```

Comment:
Is this contrary to the last guidelines that say we should use symbol packages?

Asking to know [if we need to revert this.](https://github.com/getsentry/sentry-dotnet/pull/552)

Reply:
If snupkg contains portable pdb I would think it's still a viable strategy for this. At the end of the day we just need a correlation from binary to portable pdb.

Reply:
Got it.
What is the strategy for correlating those? Mono uses mvid (module version id), on mono-symbolicate. Is this how we're doing things here too?

Reply:
Yes, both including PDB in nupkg or snupkg works. Here's the spec of CodeView Debug Directory entrye, which ties binary to the PDB: https://github.com/dotnet/runtime/blob/master/docs/design/specs/PE-COFF.md#codeview-debug-directory-entry-type-2